cmake/Modules/MacroQbtCompilerSettings.cmake

```diff
@@ -1,69 +1,42 @@
-# Sets cache variable QBT_ADDITONAL_FLAGS and QBT_ADDITONAL_CXX_FLAGS to list of additional
-# compiler flags for C and C++ (QBT_ADDITONAL_FLAGS) and for C++ only (QBT_ADDITONAL_CXX_FLAGS)
-# and appends them to CMAKE_XXX_FLAGS variables.
-
-# It could use add_compile_options(), but then it is needed to use generator expressions,
-# and most interesting of them are not compatible with Visual Studio :(
+# Compile options/definitions that should apply to all targets
+# There should be an even better way to do this, using target-centric commands
+# instead of global, directory-scope add_compile_options()
 
 macro(qbt_set_compiler_options)
-# if (NOT QBT_ADDITONAL_FLAGS)
     if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-        #-Wshadow -Wconversion ?
-        set(_GCC_COMMON_C_AND_CXX_FLAGS "-Wall -Wextra"
-            "-Wcast-qual -Wcast-align"
-            "-Winvalid-pch -Wno-long-long"
-            #"-fstack-protector-all"
-            #"-Werror -Wno-error=deprecated-declarations"
-        )
-        set(_GCC_COMMON_CXX_FLAGS "-fexceptions -frtti"
-            "-Woverloaded-virtual -Wold-style-cast"
-            "-Wnon-virtual-dtor"
-            #"-Weffc++"
-            #"-Werror -Wno-error=cpp"
-            # we should modify code to make these ones obsolete
-            #"-Wno-error=sign-conversion -Wno-error=float-equal"
+        add_compile_options(
+            -Wall
+            -Wextra
+            -Wcast-qual
+            -Wcast-align
+            -Winvalid-pch
+            -Wno-long-long
+            -fexceptions
+            -frtti
```

Comment:
I think -frtti and -fexceptions are both enabled by default.

Reply:
Indeed. Also I checked the documentation for all of the flags, and came to the following conclusions:
- `Wno-long-long` is also useless for us
- I noticed `Wnon-virtual-dtor` is part of a larger `Weffc++` suite of errors in GCC. I tried enabling that instead, but got spammed with errors. Perhaps we could benefit from the suggested fixes, but it does not seem worth it to enable `Weffc++` globally by default, so I just left `Wnon-virtual-dtor`.